test/e2e/autotls/auto_tls_test.go

```diff
@@ -222,8 +228,9 @@ func waitForCertificateReady(t *testing.T, clients *test.Clients, certName strin
 		}
 		return cert.Status.IsReady(), nil
 	}); err != nil {
-		t.Fatalf("Certificate %s is not ready: %v", certName, err)
+		return err
 	}
+	return nil
```

Comment:
return wait.Poll(10*time.Second, 300*time.Second, func() (bool, error) {
		cert, err := clients.NetworkingClient.Certificates.Get(certName, metav1.GetOptions{})
		if err != nil {
			if apierrors.IsNotFound(err) {
				t.Logf("Certificate %s has not been created: %v", certName, err)
				return false, nil
			}
			return false, err
		}
		if cert.Status.GetCondition(v1alpha1.CertificateConditionReady).IsFalse() {
			return true, fmt.Errorf("certificate %s failed with status %v", cert.Name, cert.Status)
		}
		return cert.Status.IsReady(), nil
	})

Reply:
you're right.